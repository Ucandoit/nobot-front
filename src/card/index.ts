export { default as ReserveCards } from './ReserveCards';
