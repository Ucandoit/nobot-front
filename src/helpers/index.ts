export * from './constants';
export * from './enums';
export * from './types';
export { default as useAsyncFunction } from './useAsyncFunction';
export { default as utils } from './utils';
