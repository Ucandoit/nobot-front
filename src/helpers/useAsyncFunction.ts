import { useCallback, useEffect, useState } from 'react';

const useAsyncFunction = <T>(
  asyncFunction: (...args: any[]) => Promise<T>,
  defaultValue: T,
  ...params: any[]
): [T, boolean, string | null, () => void] => {
  const [state, setState] = useState({
    value: defaultValue,
    isPending: true,
    error: null
  });

  const callFunction = useCallback(() => {
    setState(prev => ({
      ...prev,
      isPending: true
    }));
    asyncFunction
      .apply(null, params)
      .then(value => setState({ value, isPending: false, error: null }))
      .catch(error =>
        setState({
          value: defaultValue,
          isPending: false,
          error: error.toString()
        })
      );
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [asyncFunction, defaultValue, JSON.stringify(params)]);

  useEffect(() => {
    callFunction();
  }, [callFunction]);

  const { value, isPending, error } = state;
  return [value, isPending, error, callFunction];
};

export default useAsyncFunction;
