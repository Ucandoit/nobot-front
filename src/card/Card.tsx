import { Card as MuiCard, CardContent, makeStyles } from '@material-ui/core';
import React from 'react';
import { CardInfo } from '../helpers';
import actionImg from '../img/action.png';
import protectImg from '../img/protect.png';
import tradingImg from '../img/trading.png';
import untradableImg from '../img/untradable.png';

const useStyles = makeStyles(theme => ({
  card: {
    width: '6rem',
    margin: theme.spacing(1),
    textAlign: 'center',
    cursor: 'pointer'
  },
  cardContent: {
    padding: theme.spacing(1),
    '&:last-child': {
      paddingBottom: 'unset'
    }
  },
  imgWrapper: {
    position: 'relative'
  },
  trading: {
    position: 'absolute',
    left: '0',
    right: '0',
    marginLeft: 'auto',
    marginRight: 'auto'
  }
}));

interface CardProps {
  card: CardInfo;
  selectCard: (id: string) => void;
}

const Card = ({ card: { id, faceUrl, trading, inAction, untradable, protect, name }, selectCard }: CardProps) => {
  const classes = useStyles();
  return (
    <MuiCard className={classes.card} onClick={() => selectCard(id)}>
      <CardContent className={classes.cardContent}>
        <div className={classes.imgWrapper}>
          <img src={faceUrl} alt={name} title={name} />
          {trading ? (
            <img src={tradingImg} className={classes.trading} alt="trading" />
          ) : inAction ? (
            <img src={actionImg} className={classes.trading} alt="action" />
          ) : null}
        </div>
        <div>
          {untradable ? <img src={untradableImg} alt="" /> : null}
          {protect ? <img src={protectImg} alt="" /> : null}
        </div>
      </CardContent>
    </MuiCard>
  );
};

export default Card;
