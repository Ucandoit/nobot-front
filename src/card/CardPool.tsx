import { Grid } from '@material-ui/core';
import React from 'react';
import { CardFace } from '../helpers';
import CardV2 from './CardV2';

interface CardPoolProps {
  cards: CardFace[];
}

const CardPool = ({ cards }: CardPoolProps) => {
  return (
    <Grid container justify="center">
      {cards.map(card => (
        <CardV2 key={card.id} card={card} />
      ))}
    </Grid>
  );
};

export default CardPool;
