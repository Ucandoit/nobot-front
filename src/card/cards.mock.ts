import { Card } from '../helpers';

export const cards: Card[] = [
  {
    id: 3,
    number: 1001,
    name: '上杉ニャンしん',
    realName: '上杉謙信',
    rarity: '極',
    property: '火',
    cost: 3.0,
    military: '騎馬',
    job: '武士',
    star: 0,
    faceUrl: 'http://210.140.157.168/img/card/still/character_still_00034_2067324248.png',
    illustUrl: 'http://210.140.157.168/img/card/illustration/card_chara_00034_712855454.png',
    initialAtk: 97,
    initialDef: 75,
    initialSpd: 84,
    initialVir: 77,
    initialStg: 91,
    finalAtk: 197,
    finalDef: 140,
    finalSpd: 164,
    finalVir: 142,
    finalStg: 166,
    personality: '我らには毘沙門天の加護あり！',
    slogan: '越後の龍',
    history:
      '長尾為景の次男。上杉憲政から関東管領を継ぎ上杉姓となる。毘沙門天の旗を押し出して勝ち戦を重ね「越後の龍」として恐れられた。',
    trainSkills: '越後の龍, 必殺, 鼓舞, 馬沓, 伏兵看破, 一騎駆け',
    tradable: true,
    display: true
  },
  {
    id: 4,
    number: 1002,
    name: 'おミーちの方',
    realName: 'お市の方',
    rarity: '極',
    property: '水',
    cost: 3.0,
    military: '足軽',
    job: '姫',
    star: 0,
    faceUrl: 'http://210.140.157.168/img/card/still/character_still_00040_1306553348.png',
    illustUrl: 'http://210.140.157.168/img/card/illustration/card_chara_00040_488371038.png',
    initialAtk: 74,
    initialDef: 84,
    initialSpd: 82,
    initialVir: 98,
    initialStg: 73,
    finalAtk: 144,
    finalDef: 164,
    finalSpd: 162,
    finalVir: 198,
    finalStg: 143,
    personality: '猿面冠者の妻になどなりませぬ',
    slogan: '悲運の姫君',
    history:
      '信長の妹。夫の浅井長政が信長と決裂して敗死。織田家に戻り柴田勝家と再婚するが、勝家は秀吉と争って敗れ、自害。市も勝家に従った。',
    trainSkills: '小豆袋, 治療, 鼓舞, 迅速, 活法',
    tradable: true,
    display: true
  },
  {
    id: 5,
    number: 1003,
    name: '織田のぶニャが',
    realName: '織田信長',
    rarity: '極',
    property: '火',
    cost: 3.0,
    military: '鉄砲',
    job: '武士',
    star: 0,
    faceUrl: 'http://210.140.157.168/img/card/still/character_still_00050_2564820845.png',
    illustUrl: 'http://210.140.157.168/img/card/illustration/card_chara_00050_1597526338.png',
    initialAtk: 96,
    initialDef: 91,
    initialSpd: 87,
    initialVir: 62,
    initialStg: 83,
    finalAtk: 186,
    finalDef: 166,
    finalSpd: 162,
    finalVir: 137,
    finalStg: 158,
    personality: '…是非に及ばず',
    slogan: '第六天魔王',
    history:
      '織田信秀の子。尾張・美濃を平定後、「天下布武」を標榜して上洛。その後も各地を席巻し、半ば天下を握るが、明智光秀の謀叛で死去。',
    trainSkills: '第六天魔王, 雨火縄, 必殺, 早合, 伏兵看破, 三段撃ち',
    tradable: true,
    display: true
  },
  {
    id: 7,
    number: 1004,
    name: 'さニャだ幸村',
    realName: '真田幸村',
    rarity: '極',
    property: '火',
    cost: 2.5,
    military: '騎馬',
    job: '武士',
    star: 0,
    faceUrl: 'http://210.140.157.168/img/card/still/character_still_00100_142767575.png',
    illustUrl: 'http://210.140.157.168/img/card/illustration/card_chara_00100_1310624017.png',
    initialAtk: 94,
    initialDef: 70,
    initialSpd: 72,
    initialVir: 75,
    initialStg: 80,
    finalAtk: 184,
    finalDef: 115,
    finalSpd: 117,
    finalVir: 135,
    finalStg: 130,
    personality: 'さニャだ左衛門佐、 義によって参戦いたす！',
    slogan: '日本一の兵',
    history:
      '真田昌幸の次男。本名は信繁。関ヶ原戦後、豊臣家の招きで大坂へ入城。大坂夏の陣で徳川本陣に迫る奮戦を見せ「日本一の兵」と称された。',
    trainSkills: '真田丸, 一騎駆け, 馬廻衆, 突撃, 信義仁',
    tradable: true,
    display: true
  },
  {
    id: 8,
    number: 1005,
    name: 'シャムづ義弘',
    realName: '島津義弘',
    rarity: '極',
    property: '地',
    cost: 3.0,
    military: '騎馬',
    job: '武士',
    star: 0,
    faceUrl: 'http://210.140.157.168/img/card/still/character_still_00109_1467951905.png',
    illustUrl: 'http://210.140.157.168/img/card/illustration/card_chara_00109_1674371804.png',
    initialAtk: 90,
    initialDef: 93,
    initialSpd: 79,
    initialVir: 81,
    initialStg: 82,
    finalAtk: 180,
    finalDef: 193,
    finalSpd: 149,
    finalVir: 131,
    finalStg: 157,
    personality: '死にたくない者はそこをどけ！',
    slogan: '薩摩の鬼武者',
    history:
      '島津貴久の次男。兄の義久を補佐し、島津家の九州制覇に貢献した。関ヶ原では西軍に属し、敵軍中央を突破して海路帰国。惟新と号した。',
    trainSkills: undefined,
    tradable: true,
    display: true
  }
];
