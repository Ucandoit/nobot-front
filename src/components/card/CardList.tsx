import React, { useState, useEffect } from 'react';
import * as request from 'superagent';
import { Card } from '../../helpers/types';
import { Link, useLocation } from 'react-router-dom';

const rarityList = ['煌', '極', '稀', '珍', '並', '宝', '誉'];

const CardList: React.FC = () => {
  const location = useLocation<{ page: number; rarity: string }>();
  const [page, setPage] = useState(location.state ? location.state.page : 0);
  const [totalPage, setTotalPage] = useState(0);
  const [rarity, setRarity] = useState(location.state ? location.state.rarity : '');
  const [cards, setCards] = useState<Card[]>([]);

  const getCards = async () => {
    const endpoint = rarity === '' ? '?' : `/search/findCardsByRarity?rarity=${rarity}&`;
    const response = await request.get(`${ROOT_API}/api/rest/cards${endpoint}page=${page}&size=50&sort=number,asc`);
    setCards(response.body._embedded.cards);
    setTotalPage(response.body.page.totalPages);
  };

  useEffect(() => {
    getCards();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [page, rarity]);

  const pageElement = () => {
    const pages = [];
    for (let i = 0; i < totalPage; i++) {
      if (i === page) {
        pages.push(<span key={i}>{i}</span>);
      } else {
        pages.push(
          <span key={i} className="clickable" onClick={() => setPage(i)}>
            {i}
          </span>
        );
      }
    }
    return <div className="pages">{pages}</div>;
  };

  const handleRarityChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
    setRarity(event.target.value);
  };

  return (
    <div className="card-list">
      <select value={rarity} onChange={handleRarityChange}>
        <option value="">All</option>
        {rarityList.map(r => (
          <option key={r} value={r}>
            {r}
          </option>
        ))}
      </select>
      {pageElement()}
      <div className="row">
        <span style={{ width: '40px' }}>No.</span>
        <span style={{ width: '30px' }}>img</span>
        <span style={{ width: '100px' }}>name</span>
        <span style={{ width: '40px' }}>rarity</span>
        <span style={{ width: '70px' }}>property</span>
        <span style={{ width: '40px' }}>cost</span>
        <span style={{ width: '70px' }}>military</span>
        <span style={{ width: '40px' }}>job</span>
      </div>
      {cards.map(card => (
        <div className="row" key={card.id}>
          <span style={{ width: '40px' }}>{card.number}</span>
          <span style={{ width: '30px' }}>
            <img src={card.faceUrl} alt="" />
          </span>
          <span style={{ width: '100px' }}>{card.realName}</span>
          <span style={{ width: '40px' }}>
            {card.rarity}
            {getStar(card.star)}
          </span>
          <span style={{ width: '70px' }}>{card.property}</span>
          <span style={{ width: '40px' }}>{card.cost}</span>
          <span style={{ width: '70px' }}>{card.military}</span>
          <span style={{ width: '40px' }}>{card.job}</span>
          <span>
            <Link to={{ pathname: `/cards/${card.id}`, state: { page, rarity } }}>Edit</Link>
          </span>
        </div>
      ))}
    </div>
  );
};

const getStar = (star: number) => {
  if (star === 1) {
    return '\u2606';
  } else if (star === 2) {
    return '\u2606\u2606';
  } else {
    return '';
  }
};

export default CardList;
