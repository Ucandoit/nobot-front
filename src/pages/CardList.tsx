import { makeStyles, Paper, TablePagination } from '@material-ui/core';
import React, { useCallback, useState } from 'react';
import { getCards } from '../api';
import { CardTable, CardTableToolbar } from '../card';
import { usePagination, useSort } from '../commons';
import { Card, CardFilters, CardListAndCount, useAsyncFunction } from '../helpers';

const useStyles = makeStyles(theme => ({
  paper: {
    width: '100%'
  }
}));

const initialValue: CardListAndCount = [[], 0];

const CardList: React.FC = () => {
  const classes = useStyles();

  const [filters, setFilters] = useState<CardFilters>({});

  const { page, size, changePage, changeRowsPerPage } = usePagination();
  const { sort, order, changeSort } = useSort<Card>('number');

  const [[cards, count]] = useAsyncFunction<CardListAndCount>(getCards, initialValue, page, size, sort, order, filters);

  const handleChangeFilter = useCallback(
    (property: keyof CardFilters, value: string) => {
      setFilters({
        ...filters,
        [property]: value
      });
    },
    [filters]
  );

  const handleResetFilter = useCallback(() => {
    setFilters({});
  }, []);

  return (
    <>
      <Paper className={classes.paper}>
        <CardTableToolbar filters={filters} changeFilter={handleChangeFilter} resetFilter={handleResetFilter} />
        <CardTable cards={cards} sort={sort} order={order} changeSort={changeSort} />
        <TablePagination
          rowsPerPageOptions={[5, 10, 25]}
          component="div"
          count={count}
          rowsPerPage={size}
          page={page}
          onChangePage={changePage}
          onChangeRowsPerPage={changeRowsPerPage}
        />
      </Paper>
    </>
  );
};

export default CardList;
