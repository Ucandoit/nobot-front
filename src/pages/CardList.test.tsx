import { TablePagination } from '@material-ui/core';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import { CardTable, CardTableToolbar } from '../card';
import * as useAsyncFunction from '../helpers/useAsyncFunction';
import { cards } from '../mocks';
import CardList from './CardList';

describe('CardList', () => {
  let wrapper: ShallowWrapper;
  let mockUseAsyncFunction: jest.SpyInstance<[any, boolean, string | null, () => void]>;

  beforeAll(() => {
    mockUseAsyncFunction = jest.spyOn(useAsyncFunction, 'default');
    mockUseAsyncFunction.mockReturnValue([[cards, 20], false, null, jest.fn()]);
  });

  beforeEach(() => {
    wrapper = shallow(<CardList />);
  });

  afterAll(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should change page when onChangePage triggered on TablePagination', () => {
    wrapper.find(TablePagination).invoke('onChangePage')(null, 2);
    expect(mockUseAsyncFunction.mock.calls.pop()[2]).toEqual(2);
  });

  it('should change page and size when onChangeRowsPerPage triggered on TablePagination', () => {
    wrapper.find(TablePagination).invoke('onChangeRowsPerPage')!({ target: { value: '5' } } as React.ChangeEvent<
      HTMLInputElement
    >);
    expect(mockUseAsyncFunction.mock.calls.pop()[2]).toEqual(0);
    expect(mockUseAsyncFunction.mock.calls.pop()[3]).toEqual(5);
  });

  it('should change sort state when changeSort triggered on CardTable', () => {
    wrapper.find(CardTable).invoke('changeSort')('cost');
    expect(mockUseAsyncFunction.mock.calls.pop()[4]).toEqual('cost');
    expect(mockUseAsyncFunction.mock.calls.pop()[5]).toEqual('asc');

    wrapper.find(CardTable).invoke('changeSort')('cost');
    expect(mockUseAsyncFunction.mock.calls.pop()[4]).toEqual('cost');
    expect(mockUseAsyncFunction.mock.calls.pop()[5]).toEqual('desc');
  });

  it('should change filter state when changeFilter triggered', () => {
    wrapper.find(CardTableToolbar).invoke('changeFilter')('cost', '2.5');
    expect(mockUseAsyncFunction.mock.calls.pop()[6]).toEqual({
      cost: '2.5'
    });
  });

  it('should reset filter state when resetFilter triggered', () => {
    wrapper.find(CardTableToolbar).invoke('resetFilter')();
    expect(mockUseAsyncFunction.mock.calls.pop()[6]).toEqual({});
  });
});
