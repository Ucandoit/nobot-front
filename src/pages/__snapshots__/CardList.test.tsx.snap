// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList should render correctly 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Paper))
    className="makeStyles-paper-1"
  >
    <CardTableToolbar
      changeFilter={[Function]}
      filters={Object {}}
      resetFilter={[Function]}
    />
    <CardTable
      cards={
        Array [
          Object {
            "cost": 3,
            "display": true,
            "faceUrl": "http://210.140.157.168/img/card/still/character_still_00034_2067324248.png",
            "finalAtk": 197,
            "finalDef": 140,
            "finalSpd": 164,
            "finalStg": 166,
            "finalVir": 142,
            "history": "長尾為景の次男。上杉憲政から関東管領を継ぎ上杉姓となる。毘沙門天の旗を押し出して勝ち戦を重ね「越後の龍」として恐れられた。",
            "id": 3,
            "illustUrl": "http://210.140.157.168/img/card/illustration/card_chara_00034_712855454.png",
            "initialAtk": 97,
            "initialDef": 75,
            "initialSpd": 84,
            "initialStg": 91,
            "initialVir": 77,
            "job": "武士",
            "military": "騎馬",
            "name": "上杉ニャンしん",
            "number": 1001,
            "personality": "我らには毘沙門天の加護あり！",
            "property": "火",
            "rarity": "極",
            "realName": "上杉謙信",
            "slogan": "越後の龍",
            "star": 0,
            "tradable": true,
            "trainSkills": "越後の龍, 必殺, 鼓舞, 馬沓, 伏兵看破, 一騎駆け",
          },
          Object {
            "cost": 3,
            "display": true,
            "faceUrl": "http://210.140.157.168/img/card/still/character_still_00040_1306553348.png",
            "finalAtk": 144,
            "finalDef": 164,
            "finalSpd": 162,
            "finalStg": 143,
            "finalVir": 198,
            "history": "信長の妹。夫の浅井長政が信長と決裂して敗死。織田家に戻り柴田勝家と再婚するが、勝家は秀吉と争って敗れ、自害。市も勝家に従った。",
            "id": 4,
            "illustUrl": "http://210.140.157.168/img/card/illustration/card_chara_00040_488371038.png",
            "initialAtk": 74,
            "initialDef": 84,
            "initialSpd": 82,
            "initialStg": 73,
            "initialVir": 98,
            "job": "姫",
            "military": "足軽",
            "name": "おミーちの方",
            "number": 1002,
            "personality": "猿面冠者の妻になどなりませぬ",
            "property": "水",
            "rarity": "極",
            "realName": "お市の方",
            "slogan": "悲運の姫君",
            "star": 0,
            "tradable": true,
            "trainSkills": "小豆袋, 治療, 鼓舞, 迅速, 活法",
          },
          Object {
            "cost": 3,
            "display": true,
            "faceUrl": "http://210.140.157.168/img/card/still/character_still_00050_2564820845.png",
            "finalAtk": 186,
            "finalDef": 166,
            "finalSpd": 162,
            "finalStg": 158,
            "finalVir": 137,
            "history": "織田信秀の子。尾張・美濃を平定後、「天下布武」を標榜して上洛。その後も各地を席巻し、半ば天下を握るが、明智光秀の謀叛で死去。",
            "id": 5,
            "illustUrl": "http://210.140.157.168/img/card/illustration/card_chara_00050_1597526338.png",
            "initialAtk": 96,
            "initialDef": 91,
            "initialSpd": 87,
            "initialStg": 83,
            "initialVir": 62,
            "job": "武士",
            "military": "鉄砲",
            "name": "織田のぶニャが",
            "number": 1003,
            "personality": "…是非に及ばず",
            "property": "火",
            "rarity": "極",
            "realName": "織田信長",
            "slogan": "第六天魔王",
            "star": 0,
            "tradable": true,
            "trainSkills": "第六天魔王, 雨火縄, 必殺, 早合, 伏兵看破, 三段撃ち",
          },
          Object {
            "cost": 2.5,
            "display": true,
            "faceUrl": "http://210.140.157.168/img/card/still/character_still_00100_142767575.png",
            "finalAtk": 184,
            "finalDef": 115,
            "finalSpd": 117,
            "finalStg": 130,
            "finalVir": 135,
            "history": "真田昌幸の次男。本名は信繁。関ヶ原戦後、豊臣家の招きで大坂へ入城。大坂夏の陣で徳川本陣に迫る奮戦を見せ「日本一の兵」と称された。",
            "id": 7,
            "illustUrl": "http://210.140.157.168/img/card/illustration/card_chara_00100_1310624017.png",
            "initialAtk": 94,
            "initialDef": 70,
            "initialSpd": 72,
            "initialStg": 80,
            "initialVir": 75,
            "job": "武士",
            "military": "騎馬",
            "name": "さニャだ幸村",
            "number": 1004,
            "personality": "さニャだ左衛門佐、 義によって参戦いたす！",
            "property": "火",
            "rarity": "極",
            "realName": "真田幸村",
            "slogan": "日本一の兵",
            "star": 0,
            "tradable": true,
            "trainSkills": "真田丸, 一騎駆け, 馬廻衆, 突撃, 信義仁",
          },
          Object {
            "cost": 3,
            "display": true,
            "faceUrl": "http://210.140.157.168/img/card/still/character_still_00109_1467951905.png",
            "finalAtk": 180,
            "finalDef": 193,
            "finalSpd": 149,
            "finalStg": 157,
            "finalVir": 131,
            "history": "島津貴久の次男。兄の義久を補佐し、島津家の九州制覇に貢献した。関ヶ原では西軍に属し、敵軍中央を突破して海路帰国。惟新と号した。",
            "id": 8,
            "illustUrl": "http://210.140.157.168/img/card/illustration/card_chara_00109_1674371804.png",
            "initialAtk": 90,
            "initialDef": 93,
            "initialSpd": 79,
            "initialStg": 82,
            "initialVir": 81,
            "job": "武士",
            "military": "騎馬",
            "name": "シャムづ義弘",
            "number": 1005,
            "personality": "死にたくない者はそこをどけ！",
            "property": "地",
            "rarity": "極",
            "realName": "島津義弘",
            "slogan": "薩摩の鬼武者",
            "star": 0,
            "tradable": true,
            "trainSkills": undefined,
          },
        ]
      }
      changeSort={[Function]}
      order="asc"
      sort="number"
    />
    <WithStyles(ForwardRef(TablePagination))
      component="div"
      count={20}
      onChangePage={[Function]}
      onChangeRowsPerPage={[Function]}
      page={0}
      rowsPerPage={10}
      rowsPerPageOptions={
        Array [
          5,
          10,
          25,
        ]
      }
    />
  </WithStyles(ForwardRef(Paper))>
</Fragment>
`;
